File: app/src/main/java/com/katkov/lolachievements/data/local/dao/SummonerDao.java
package com.katkov.lolachievements.data.local.dao;

import com.katkov.lolachievements.data.local.model.Summoner;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Single;

@Dao
public interface SummonerDao {

    @Insert
    void insert(Summoner summoner);

    @Update
    void update(Summoner summoner);

    @Delete
    void delete(Summoner summoner);

    @Query("SELECT * from summoner")
    Single<Summoner> getAll();
}

File: app/src/main/java/com/katkov/lolachievements/data/local/dao/SummonerDao.kt
package com.katkov.lolachievements.data.local.dao

import androidx.room.*
import com.katkov.lolachievements.data.local.model.SummonerDbModel
import io.reactivex.Single

@Dao
interface SummonerDao {

    @Query("SELECT COUNT(*) FROM SummonerDbModel")
    fun getRowsCount(): Single<Int>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(summonerDbModel: SummonerDbModel)

    @Query("SELECT * FROM SummonerDbModel")
    fun getAll(): Single<SummonerDbModel>

    @Update
    fun update(summonerDbModel: SummonerDbModel)

    @Delete
    fun delete(summonerDbModel: SummonerDbModel)
}
